homeassistant/components/izone/discovery.py

```diff
@@ -15,15 +16,17 @@
     DISPATCH_ZONE_UPDATE,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
 
 class DiscoveryService(pizone.Listener):
     """Discovery data and interfacing with pizone library."""
 
-    def __init__(self, hass):
+    def __init__(self, hass: HomeAssistant) -> None:
         """Initialise discovery service."""
         super().__init__()
         self.hass = hass
-        self.pi_disco = None
+        self.pi_disco = None  # type: pizone.DiscoveryService | None
```

Comment:
You should set the type correctly, not as a comment:
```suggestion
        self.pi_disco: pizone.DiscoveryService | None = None
```